### PR TITLE
Move client API requests into dedicated service

### PR DESCRIPTION
**Pull request by dotproto**

### CODE REVIEW

**Comment by kelsin** [comment]
getIssue needs to pass in an id, and the service needs to reference opt, not params.

File: client/scripts/app.js [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+  //   owner: 'name of the user/org that owns the repo',
+  //   repo: 'name of the repository'
+  // }
+  this.getIssues = function(opts, onSuccess) {
+    $http({ method: 'GET', url: '/api/' + opts.owner + '/' + opts.repo + '/issues' })
+      .success(onSuccess);
+  }
+
+  this.getIssue = function(opts, onSuccess) {
+    $http({ method: 'GET', url: '/api/' + params.owner + '/' + params.repo + '/issue/' + params.id })
```
Comment (mschnee) [inline client/scripts/app.js]: params is undefined, did you mean opts?

File: client/scripts/app.js [hunk 2]
```diff
@@ -20,29 +36,34 @@ gimli.config(function($routeProvider, $locationProvider) {
...
+gimli.controller('IssueController', ['$scope', '$routeParams', 'GimliApiService',
+  function($scope, $routeParams, gimliApi){
     $scope.issue = {};
     params = $routeParams;
 
-    $http({ method: 'GET', url: '/api/' + params.owner + '/' + params.repo + '/issue/' + params.id })
-      .success(function(data, status, headers, config) {
+    gimliApi.getIssue({
+        owner: params.owner,
+        repo: params.repo
```
Comment (mschnee) [inline client/scripts/app.js]: missing: 

```
id: params.id
```

**Comment by dotproto** [comment]
This commit should address the issues identified by @Kelsin and @mschnee. That'll teach me to PR right before I go to bed.

**Pull request merged**